pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -118,65 +140,346 @@ func (bm *backupManager) Sync(backup *v1alpha1.NebulaBackup) error {
+
+
+# Step 3: remove backups that need to be removed
+echo "Removing previous expired backups..."
+max_reserved_time_epoach=$(convert_to_sec $max_reserved_time)
+if [[ $max_reserved_time_epoach -gt 0 ]]; then
+    echo "Maximum retention time of $max_reserved_time set. Removing all previous backups exceeding this retention time...."
+    now=$(date +"%s")
+    echo "Maximum retention time epoach: $max_reserved_time_epoach"
+    echo "Current time epoach: $now"
+    for ind in ${!backup_names[@]}
+    do
+        nb_date=$(echo "${backup_dates[ind]}" | tr "T" " " | xargs)
+        nb_date_epoach=$(date -d "$nb_date" +"%s")
+        diff=$((now - nb_date_epoach))
+        echo "Backing up file ${backup_names[ind]}. Backup date: \"$nb_date ($nb_date_epoach)\". Diff: \"$diff\""
+
+        if [[ $diff -gt $max_reserved_time_epoach ]]; then
+            echo "File ${backup_names[ind]} is older than the maximum reserved time. Deleting..."
+            br-ent cleanup --meta "$meta_address" --storage "$storage" --s3.access_key $ACCESS_KEY --s3.secret_key $SECRET_KEY --s3.region "$s3_region" --s3.endpoint "$s3_endpoint" --name "${backup_names[ind]}"
+        fi
+    done
+    echo "All necessary previous backups cleaned up."
+elif [[ $max_backups -gt 0 ]]; then
+    if [[ $total_backups -gt $max_backups ]]; then
+        echo "Maximum number of backups $max_backups set. Removing all backups exceeding this number starting with the oldest backup..."
+        num_to_del=$((total_backups - max_backups - 1))
+        echo "Number of previous backups to delete: $num_to_del"
+        for ind in $(seq 0 $num_to_del)
+        do
+            br-ent cleanup --meta "$meta_address" --storage "$storage" --s3.access_key $ACCESS_KEY --s3.secret_key $SECRET_KEY --s3.region "$s3_region" --s3.endpoint "$s3_endpoint" --name "${backup_names[ind]}"
+        done
+        echo "All necessary previous backups cleaned up."
+    else
+        echo "Current number of backups has not exceeded the maximum number of backups to be kept. Will leave all previous backups alone."
+    fi
+else
+    echo "No max retention time or maximum number of backups set. Will leave all previous backups alone"
+fi
+`,
+		},
+	}
+}
+
+func (bm *backupManager) generateBackupJob(backup *v1alpha1.NebulaBackup, metaAddr string) *batchv1.Job {
+	if backup.Spec.MaxReservedTime != nil || backup.Spec.MaxBackups != nil && *backup.Spec.MaxBackups != int32(0) {
+		maxReservedTimeToUse := "0"
+		if backup.Spec.MaxReservedTime != nil {
+			maxReservedTimeToUse = *backup.Spec.MaxReservedTime
+		}
+
+		maxBackupToUse := int32(0)
+		if backup.Spec.MaxBackups != nil {
+			maxBackupToUse = *backup.Spec.MaxBackups
+		}
+
+		backupCmd := fmt.Sprintf("/usr/local/bin/runnable/backup-cleanup-run.sh -m %v -s s3://%v -r %v -e %v -b %v -t %v",
+			metaAddr, backup.Spec.BR.S3.Bucket, backup.Spec.BR.S3.Region, backup.Spec.BR.S3.Endpoint, maxBackupToUse, maxReservedTimeToUse,
+		)
+		return &batchv1.Job{
```

Comment:
Here OwnerReference is lost

Reply:
@MegaByte875 Got it, will add a owner reference when creating this.

Reply:
Fixed